src/WebJobs.Script/Host/WorkerFunctionMetadataProvider.cs

```diff
@@ -141,9 +142,13 @@ public async Task<FunctionMetadataResult> GetFunctionMetadataAsync(IEnumerable<R
                         {
                             _logger.LogWarning(ex, "Removing errored webhost language worker channel for runtime: {workerRuntime} workerId:{workerId}", _workerRuntime, workerId);
                             await _channelManager.ShutdownChannelIfExistsAsync(_workerRuntime, workerId, ex);
+                            errors ??= [];
+                            errors.Add(ex);
```

Comment:
This behavior I am unsure of keeping. Not sure what the expected behavior is when a worker fails and it is supposed to perform worker indexing? Take a bad python app missing `functions_app.py`. The worker will start and immediately exit with an error.

What should our function indexing behavior be? Without this change the experience is we proceed with no errors and no functions found. With this change we will fault the script host.

Reply:
Chatted with @fabiocav - sounds like the right change to make.